doc/musig-spec.mediawiki

```diff
@@ -184,64 +318,64 @@ Input:
+* The public key of the signer ''pk<sup>*</sup>'' (in ''pk<sub>1..u</sub>'' of the session_ctx''): a 32-byte array
+* The ''session_ctx'': a [[#session-context|Session Context]] data structure
 
-The algorithm ''PartialSigVerifyInternal(psig, pubnonce, aggnonce, pk<sub>1..u</sub>, pk<sup>*</sup>, m)'' is defined as:
+The algorithm ''PartialSigVerifyInternal(psig, pubnonce, pk<sup>*</sup>, session_ctx)'' is defined as:
+* Let ''(Q, g, _, b, R, e) = GetSessionValues(session_ctx)''; fail if that fails
 * Let ''s = int(psig)''; fail if ''s &ge; n''
-* Let ''R<sub>1</sub> = pointc(aggnonce[0:33]), R<sub>2</sub> = pointc(aggnonce[33:66])''; fail if that fails
-* Let ''Q = KeyAggInternal(pk<sub>1..u</sub>)''; fail if that fails
-* Let ''b = int(hash<sub>MuSig/noncecoef</sub>(aggnonce || bytes(Q) || m)) mod n''
-* Let ''R = R<sub>1</sub> + b⋅R<sub>2</sub>''
 * Let ''R<sup>*</sup><sub>1</sub> = pointc(pubnonce[0:33]), R<sup>*</sup><sub>2</sub> = pointc(pubnonce[33:66])''
 * Let ''R<sup>*</sup>' = R<sup>*</sup><sub>1</sub> + b⋅R<sup>*</sup><sub>2</sub>''
 * Let ''R<sup>*</sup> = R<sup>*</sup>' '' if ''has_even_y(R)'', otherwise let ''R<sup>*</sup> = -R<sup>*</sup>' ''
-* Let ''e = int(hash<sub>BIP0340/challenge</sub>(bytes(R) || bytes(Q) || m)) mod n''
-* Let ''mu = KeyAggCoeff(pk<sub>1..u</sub>, pk<sup>*</sup>)''
-* Let ''P' = point(pk<sup>*</sup>)''; fail if that fails
-* Let ''P = P' '' if ''has_even_y(Q)'', otherwise let ''P = -P' ''
-* Fail if ''s⋅G &ne; R<sup>*</sup> + e⋅mu⋅P''
+* Let ''g<sub>v</sub> = 1'' if ''has_even_y(Q)'', otherwise let ''g<sub>v</sub> = -1''
+* Let ''g' = g<sub>v</sub>⋅g mod n''
+* <div id="SigVerify negation"></div>Let ''P = g'⋅point(pk<sup>*</sup>)''; fail if that fails (See [[#negation-of-the-public-key-when-verifying|Negation Of The Public Key When Verifying]])
+* Let ''a = GetSessionKeyAggCoeff(session_ctx, P)''; fail if that fails
+* Fail if ''s⋅G &ne; R<sup>*</sup> + e⋅a⋅P''
 * Return success iff no failure occurred before reaching this point.
 
 ==== Partial Signature Aggregation ====
 
 Input:
-* The final nonce ''R'' as created during  ''Sign'' or ''PartialSigVerify'': a point
 * The number ''u'' of signatures with ''0 < u < 2^32''
-* The partial signatures ''sig<sub>1..u</sub>'': ''u'' 32-byte arrays
+* The partial signatures ''psig<sub>1..u</sub>'': ''u'' 32-byte arrays
+* The ''session_ctx'': a [[#session-context|Session Context]] data structure
 
-The algorithm ''SigAgg(R, sig<sub>1..u</sub>)'' is defined as:
+The algorithm ''PartialSigAgg(psig<sub>1..u</sub>, session_ctx)'' is defined as:
+* Let ''(Q, _, t, _, _, R, e) = GetSessionValues(session_ctx)''; fail if that fails
 * For ''i = 1 .. u'':
-** Let ''s<sub>i</sub> = int(sig<sub>i</sub>)''; fail if ''s<sub>i</sub> &ge; n''.
-* Let ''s = s<sub>1</sub> + ... + s<sub>u</sub> mod n''
+** Let ''s<sub>i</sub> = int(psig<sub>i</sub>)''; fail if ''s<sub>i</sub> &ge; n''.
+* Let ''g<sub>v</sub> = 1'' if ''has_even_y(Q)'', otherwise let ''g<sub>v</sub> = -1''
```

Comment:
Not sure how much this matters, but this is a notable change in requirement for the partial signature aggregator role. It now needs to have access to the XY aggregate public key.

Reply:
> New way better.

Yeah agreed. In the secp-zkp implementation we do not require the final pubkey in aggregation but instead store the final (parity-included) tweak in the session object (which is a subset of the session datastructure of the spec).